lib/ansible/module_utils/urls.py

```diff
@@ -302,6 +302,16 @@ def match_hostname(cert, hostname):
 
     HAS_MATCH_HOSTNAME = True
 
+if hasattr(socket, 'create_connection'):
+    create_connection = socket.create_connection
+else:
+    # Just a graceful fallback for py < 2.6, implementing old behaviour.
+    # Wouldn't work with IPv6, but it is unlikely that python 2.4 could be met
+    # on IPv6-only machine.
+    def create_connection(address):
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sock.connect(address)
+        return sock
```

Comment:
Handling exception each time on python2.4 definitely would not be the faster way. The try/except block also has its own little overhead, so it would be slower even on python>2.4. Though I don't think it should be bother someone at all.